test/directory_watcher/shared.dart

```diff
@@ -2,270 +2,241 @@
+    await startWatcher(path: "dir");
 
     renameDir("dir", "moved_dir");
     createDir("dir");
-    inAnyOrder([isRemoveEvent("dir/a.txt"), isRemoveEvent("dir/b.txt")]);
+    await inAnyOrder([isRemoveEvent("dir/a.txt"), isRemoveEvent("dir/b.txt")]);
   });
 
   // Regression test for b/30768513.
   test(
       "doesn't crash when the directory is moved immediately after a subdir "
-      "is added", () {
+      "is added", () async {
     writeFile("dir/a.txt");
     writeFile("dir/b.txt");
 
-    startWatcher(path: "dir");
+    await startWatcher(path: "dir");
 
     createDir("dir/subdir");
     renameDir("dir", "moved_dir");
     createDir("dir");
-    inAnyOrder([isRemoveEvent("dir/a.txt"), isRemoveEvent("dir/b.txt")]);
+    await inAnyOrder([isRemoveEvent("dir/a.txt"), isRemoveEvent("dir/b.txt")]);
   });
 
   group("moves", () {
-    test('notifies when a file is moved within the watched directory', () {
+    test('notifies when a file is moved within the watched directory',
+        () async {
       writeFile("old.txt");
-      startWatcher();
+      await startWatcher();
       renameFile("old.txt", "new.txt");
 
-      inAnyOrder([isAddEvent("new.txt"), isRemoveEvent("old.txt")]);
+      await inAnyOrder([isAddEvent("new.txt"), isRemoveEvent("old.txt")]);
     });
 
     test('notifies when a file is moved from outside the watched directory',
-        () {
+        () async {
       writeFile("old.txt");
       createDir("dir");
-      startWatcher(path: "dir");
+      await startWatcher(path: "dir");
 
       renameFile("old.txt", "dir/new.txt");
       expectAddEvent("dir/new.txt");
     });
 
-    test('notifies when a file is moved outside the watched directory', () {
+    test('notifies when a file is moved outside the watched directory',
+        () async {
       writeFile("dir/old.txt");
-      startWatcher(path: "dir");
+      await startWatcher(path: "dir");
 
       renameFile("dir/old.txt", "new.txt");
       expectRemoveEvent("dir/old.txt");
     });
 
-    test('notifies when a file is moved onto an existing one', () {
+    test('notifies when a file is moved onto an existing one', () async {
       writeFile("from.txt");
       writeFile("to.txt");
-      startWatcher();
+      await startWatcher();
 
       renameFile("from.txt", "to.txt");
-      inAnyOrder([isRemoveEvent("from.txt"), isModifyEvent("to.txt")]);
+      await inAnyOrder([isRemoveEvent("from.txt"), isModifyEvent("to.txt")]);
     });
   });
 
-  // Most of the time, when multiple filesystem actions happen in sequence,
-  // they'll be batched together and the watcher will see them all at once.
-  // These tests verify that the watcher normalizes and combine these events
-  // properly. However, very occasionally the events will be reported in
-  // separate batches, and the watcher will report them as though they occurred
-  // far apart in time, so each of these tests has a "backup case" to allow for
-  // that as well.
```

Comment:
Why aren't we doing this anymore?

Reply:
This doesn't seem to be an issue. I have yet to run into the 'backup case'. Also with the new test model, I'm not sure how I would even implement backup expects.

Reply:
See above—backup cases are important, and you can't reliably tell whether they're relevant with local testing. These systems are *extremely* contingent on opaque OS specifics, which is why the tests have so many alternate cases in the first place.

Reply:
Added back the backup cases.